### PR TITLE
Rtmp read pause timer - Inputstream API 2.2.0

### PR DESCRIPTION
**Pull request by phunkyfish**
v3.0.3
- Remove pause and add read pause detect timer for Inputstream API 2.2.0
- Add debug logger
- add build script for mac if using xbmc source dir

Do not merge yet

https://github.com/xbmc/xbmc/pull/17633

### CODE REVIEW

File: src/timer/Timer.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  std::function<void()> m_callback;
+  uint32_t m_timeout;
+  bool m_interval;
+  uint32_t m_endTime;
+  CEvent m_eventTimeout;
+  CTimeout m_timer;
+  std::thread m_thread;
+  bool m_running = false;
+  bool m_stopping = false;
```
Comment (AlwinEsch) [inline src/timer/Timer.h]: This is nowhere used, maybe wrongly added?

File: src/utils/Log.h [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      return "LOG_SEVERE:  ";
+    case ADDON_LOG_FATAL:
+      return "LOG_FATAL:   ";
+    default:
+      break;
+  }
+  return "LOG_UNKNOWN: ";
+}
+
+inline void kodiLog(const AddonLog logLevel, const char* format, ...)
```
Comment (AlwinEsch) [inline src/utils/Log.h]: Has you added for later use? Seen the added log call is still by `kodi::Log`.

Has used this function on vfs.rar to make the debuging more easy, as I not need to look on Kodi log all the time.

**Review by AlwinEsch — COMMENTED** [review]
Apart from the two little things, it looks good for me and the runtime test under Linux was OK.

File: src/utils/Log.h (hunk 2 above)
Comment (phunkyfish) [reply src/utils/Log.h]: I forgot to remove it. Oops.

File: src/timer/Timer.h (hunk 1 above)
Comment (phunkyfish) [reply src/timer/Timer.h]: Will remove.

File: src/utils/Log.h (hunk 2 above)
Comment (phunkyfish) [reply src/utils/Log.h]: Or maybe I will change it to this. It’s a good example of a custom logger.

**Pull request merged**